docs/proposal/aws-resource-handling.md

```diff
@@ -0,0 +1,203 @@
+4. Yes - the next attempt to create the resource will return the already created resource.
+
+### Resources that require separate tagging without client token support
+
+#### Create - option 1
+
+- Create resource
+- Update cluster/machine object config and status
+  - If update fails attempt delete of created resource
+    - If delete fails log delete failure and return non-retryable error
+- Tag AWS resource
+- Enqueue cluster/machine update if not already available/ready or tagging fails
+
+TODO: flowchart
+
+#### Update - option 1
+
+- Query resource by ID
+- tag resource if missing tags
+- Update object status
+- Enqueue cluster/machine update if not already available/ready
+
+#### Edge case coverage - option 1
+
+1. Yes - Since the resource ID is already recorded, the update process will reconcile missing tags
+2. Yes, with caveat - If the object update fails, we attempt to rollback the creation but edge case 3 comes into play
+3. Minor mitigation - If we fail to delete the resource, we will still orphan the resource, but output a log message for querying/followup and return a non-retryable error
+4. Minor mitigation - If the process dies before recording the ID the resource is orphaned. If the process dies after recording the ID, but before tagging it is reconciled through update.
+
+#### Create - option 2
+
+- Query resource by tags to determine if resource already exists
+- Create the resource if it doesn't already exist
+- Update cluster/machine object config and status
+  - Note failure but do not return error yet
+- Tag AWS resource if needed
+  - If both update and tagging fails, delete resource
+    - If delete fails log failure prominently, return non-retryable error
+  - If only tagging fails, return retryable error
+- If update failed, return retryable error
+- Enqueue cluster/machine update if not already available/ready or tagging fails
+
+TODO: flowchart
+
+#### Update - option 2
+
+- Query resource by ID
+- Update object status
+- Enqueue cluster/machine update if not already available/ready
+
+#### Edge case coverage - option 2
+
+1. Yes - If only the tagging fails, then we will reconcile tags on update. If the update also fails, then we attempt to delete the resource.
+2. Yes, with caveat - If only the object update fails, then we throw a retryable error that will requeue the create operation and attempt to update the object after discovering the existing resource. If the tagging fails as well, then we attempt to delete the resource and edge case 3 will still apply.
+3. Minor mitigation - If we fail to delete the resource, we will still orphan the resource, but output a log message for querying/followup and return a non-retryable error
+4. Minor mitigation - If the process dies before recording the ID the resource is orphaned. If the process dies after recording the ID, but before tagging it is reconciled through update.
+
+### Resources without tag support with client token support
```

Comment:
this is a redundant case with case: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/66/files#diff-ca0d5b46102c7cf8ceb425bb9f57cd95R76

Reply:
I need to improve the wording here, the other case are resources that support tags, just not tagging on create.

Reply:
I updated the wording on the other workflow to hopefully make it more clear.